sequencer/src/state.rs

```diff
@@ -555,18 +575,12 @@ impl ValidatedState {
         // Ensure merkle tree has frontier
         if self.need_to_fetch_blocks_mt_frontier() {
             tracing::warn!("fetching block frontier from peers");
-            instance
-                .peers
-                .as_ref()
+
+            // Unwrapping here is okay as we retry until we get the proof or until the task is canceled.
+            catchup
```

Comment:
I guess technically this is not the case anymore, since this same function is also called with the Postgres catchup, in which we might actually return an `Err` (e.g. if we lose connection to the database). What we should probably do is just propagate the error here  (this function returns `Result` anyway) and then this unwrap will happen in `validate_and_apply_header` in the `ValidatedState` trait, because at _that_ point we are only dealing with the infallible `StatePeers` implementation of catchup.

Alternatively, we could in turn loop `validate_and_apply_header` forever, because why not: we get rid of a potential panic and HotShot will eventually cancel the task anyways